dotCMS/src/main/java/com/dotcms/rest/api/v1/maintenance/MaintenanceResource.java

```diff
@@ -101,29 +115,86 @@ public final Response downloadLogFile(@Context final HttpServletRequest request,
-            if(!logFile.exists()){
-                throw new DoesNotExistException("Requested LogFile: " + logFile.getCanonicalPath() + " does not exist.");
-            }
-
-            Logger.info(this.getClass(), "Requested logFile: " + logFile.getCanonicalPath());
-
-            response.setHeader( "Content-Disposition", "attachment; filename=" + fileName );
-            return Response.ok(logFile, MediaType.APPLICATION_OCTET_STREAM).build();
+        assertBackendUser(request, response);
+
+        String tailLogFolder = Config
+                .getStringProperty("TAIL_LOG_LOG_FOLDER", "./dotsecure/logs/");
+        if (!tailLogFolder.endsWith(File.separator)) {
+            tailLogFolder = tailLogFolder + File.separator;
+        }
+
+        final File logFile = new File(FileUtil.getAbsolutlePath(tailLogFolder + fileName));
+        if(!logFile.exists()){
+            throw new DoesNotExistException("Requested LogFile: " + logFile.getCanonicalPath() + " does not exist.");
+        }
+
+        Logger.info(this.getClass(), "Requested logFile: " + logFile.getCanonicalPath());
+
+        response.setHeader( "Content-Disposition", "attachment; filename=" + fileName );
+        return Response.ok(logFile, MediaType.APPLICATION_OCTET_STREAM).build();
+    }
+
+    /**
+     * This method attempts to send resolved DB dump file using an octet stream http response.
+     *
+     * @param request  http request
+     * @param response http response
+     * @return octet stream response with file contents
+     * @throws IOException
+     */
+    @Path("/_downloadDb")
+    @GET
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_OCTET_STREAM, MediaType.APPLICATION_JSON})
+    public final Response downloadDb(@Context final HttpServletRequest request,
+                                     @Context final HttpServletResponse response) throws IOException {
+        assertBackendUser(request, response);
+
+        final File dbFile = DbExporterUtil.exportToFile();
+        Logger.info(this.getClass(), "Requested dbFile: " + dbFile.getCanonicalPath());
+
+        response.setHeader("Content-Disposition", "attachment; filename=" + dbFile.getName());
+        return Response.ok(dbFile, MediaType.APPLICATION_OCTET_STREAM).build();
+    }
+
+    /**
+     * This method attempts to send resolved DB dump file using an octet stream http response.
+     *
+     * @param request  http request
+     * @param response http response
+     * @return octet stream response with file contents
+     * @throws IOException
+     */
+    @Path("/_assets")
+    @GET
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_OCTET_STREAM, MediaType.APPLICATION_JSON})
+    public final Response downloadAssets(@Context final HttpServletRequest request,
+                                         @Context final HttpServletResponse response) throws IOException {
+        final User user = Try.of(() -> assertBackendUser(request, response).getUser()).get();
```

Comment:
Downloading the assets needs to be admin only - not just maintenance portlet access.

Reply:
Done.